src/client.rs

```diff
@@ -29,6 +32,10 @@ impl ChatGPT {
         Self::new_with_config(api_key, ModelConfiguration::default())
     }
 
+     /// Constructs a new ChatGPT API client with provided API key and default configuration
+     pub fn new_with_proxy<S: Into<String>>(api_key: S,proxy: Proxy) -> crate::Result<Self> {
+        Self::new_with_proxy_andconfig(api_key, ModelConfiguration::default(),proxy)
```

Comment:
See above about naming change:
```suggestion
        Self::new_with_config_proxy(api_key, ModelConfiguration::default(), proxy)
```